app/resources/views/pages/location_search/spreadsheet.blade.php

```diff
@@ -16,7 +16,15 @@ class="form-control" id="distance"
 	</div>
 	@if (count($locations) === 0)
 		<h3>No location found matching the specified keywords</h3>
-		<a class="btn btn-default" href="/locations-added-by-me">Add New Location</a>
+			@if ($base_user->isSignedIn())
+				<a class="btn btn-default" href="/add-location">Add New Location</a>
+			@else
+				<a class="btn btn-default" href="/sign-in">Sign In</a>
```

Comment:
Did you test this link?  It looks like "/signin" is what you want without the hyphen.